rbd/mirror.go

```diff
@@ -514,3 +514,72 @@ func GetMirrorSiteName(conn *rados.Conn) (string, error) {
 	// the C code sets the size including null byte
 	return string(buf[:cSize-1]), nil
 }
+
+// CreateMirrorPeerBootstrapToken returns a token value, representing the
+// cluster and pool associated with the given IO context,  that can be provided
+// to ImportMirrorPeerBootstrapToken in order to set up mirroring between
+// pools.
+//
+// Implements:
+//  int rbd_mirror_peer_bootstrap_create(
+//    rados_ioctx_t io_ctx, char *token, size_t *max_len);
+func CreateMirrorPeerBootstrapToken(ioctx *rados.IOContext) ([]byte, error) {
```

Comment:
I believe returning the token as a string is more natural here, like for GetMirrorSiteName above.

Reply:
hehe. I went back and forth a bit on this myself and decided that because the token was supposed to be opaque it would be simpler to "round tip" and/or output to a file as bytes.
How strongly do you feel about this?

Reply:
I don't think the token is opaque. It's clearly a Base64 string, that is meant to be inserted in json files.

Reply:
Or does rbd_mirror_peer_bootstrap_create() return raw data and only the cli tool is converting it to Base64?

Reply:
The python library also returns a string.

Reply:
Maybe introduce a `MirrorPeerToken` type? It can have a `.String()` method and all if that is what users need later on.

Reply:
But why? It is definitely a Base64 string. Actually even the encoded data is a string as well, because the token is a Base64v encoded json object. So it would be a string in any case. See https://github.com/ceph/ceph/blob/a67d1cf2a7a4031609a5d37baa01ffdfef80e993/src/librbd/api/Mirror.cc#L1316-L1318

Reply:
I guess the question is if users are have a need to inspect the contents of the token. If the answer is *no*, then I do not see the need to provide details about it. In case the token needs to be exportable and consumed by an other application, then it makes sense to have it as a string.

Reply:
I have changed it to a string now. I was always on the fence about it, so it's no big deal to change. PTAL